packages/react-components/src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,270 @@
+import * as React from 'react';
+import { CSSTransition } from 'react-transition-group';
+import cx from 'clsx';
+import { css } from '@emotion/css';
+import {
+  useFloating,
+  arrow,
+  Placement,
+  autoUpdate,
+  flip,
+  offset,
+  VirtualElement,
+} from '@floating-ui/react-dom';
+
+import styles from './Tooltip.module.scss';
+
+export interface ITooltipProps {
+  children?: React.ReactNode;
+  className?: string;
+  theme?: 'invert' | 'important' | undefined;
+  placement?: Placement;
+  isVisible?: boolean;
+  withFadeAnimation?: boolean;
+  transitionDuration?: number;
+  transitionDelay?: number;
+  hoverOutDelayTimeout?: number;
+  offsetMainAxis?: number;
+  triggerOnClick?: boolean;
+  arrowOffsetY?: number;
+  arrowOffsetX?: number;
+  triggerRenderer: () => React.ReactNode;
+  referenceElement?: VirtualElement;
+  onOpen?: () => void;
+  onClose?: () => void;
+}
+
+const sleep = (milliseconds: number) => {
+  return new Promise((resolve) => setTimeout(resolve, milliseconds));
+};
+
+const baseClass = 'tooltip';
+
+export const Tooltip: React.FC<ITooltipProps> = (props) => {
+  const {
+    triggerRenderer,
+    referenceElement,
+    children,
+    className,
+    theme,
+    placement,
+    isVisible,
+    withFadeAnimation = true,
+    transitionDuration = 200,
+    transitionDelay = 0,
+    hoverOutDelayTimeout = 100,
+    offsetMainAxis = 8,
+    triggerOnClick = false,
+    arrowOffsetY,
+    arrowOffsetX,
+    onOpen,
+    onClose,
+  } = props;
+
+  const isManaged = isVisible !== undefined;
+  const arrowRef = React.useRef<HTMLDivElement | null>(null);
+  const [visible, setVisibility] = React.useState(isVisible);
+  const isHovered = React.useRef(false);
+
+  const {
+    x,
+    y,
+    reference,
+    floating,
+    strategy,
+    update,
+    refs,
+    placement: updatedPlacement,
+    middlewareData: { arrow: { x: arrowX, y: arrowY } = {} },
+  } = useFloating({
+    middleware: [
+      offset({ mainAxis: offsetMainAxis }),
+      arrow({ element: arrowRef }),
+      flip(),
+    ],
+    placement: placement,
+  });
+
+  React.useEffect(() => {
+    referenceElement && reference(referenceElement);
+  }, [reference, referenceElement]);
+
+  React.useEffect(() => {
+    setVisibility(isVisible);
+  }, [isVisible]);
+
+  React.useEffect(() => {
+    document.addEventListener('keydown', handleHideOnEscape);
+    return () => {
+      document.removeEventListener('keydown', handleHideOnEscape);
+    };
+  }, []);
+
+  React.useEffect(() => {
+    if (!refs.reference.current || !refs.floating.current) {
+      return;
+    }
+
+    // Only call this when the floating element is rendered
+    return autoUpdate(refs.reference.current, refs.floating.current, update);
+  }, [refs.reference, refs.floating, update, updatedPlacement]);
```

Comment:
This effect needs to rely on `visible` state value even if it is not directly used here because it synchronizes the visibility with the tooltip position. The "refs" are React Refs, which means their values are not reactive; hence the change is not triggering rerender and replying to the effect. This causes the bug when the tooltip is not opened initially as it is not properly tracking the element on the screen.
```suggestion
  }, [refs.reference, refs.floating, update, updatedPlacement, visible]);
```

Without the change:
<video src="https://user-images.githubusercontent.com/9700566/169988307-4674d39e-2a26-47c0-a334-544f49aa2e4c.mov" />

After syncing with `visible` state:
<video src="https://user-images.githubusercontent.com/9700566/169988426-0149eac0-7ba6-4605-80c4-47260e81964f.mov" />

Reply:
thanks!